src/components/lib/sl/sl.c

```diff
@@ -614,7 +617,7 @@ sl_sched_loop(void)
 			 * states of it's child threads) and normal notifications (mainly activations from
 			 * it's parent scheduler).
 			 */
-			pending = cos_sched_rcv(g->sched_rcv, RCV_ALL_PENDING, timeout,
+			pending = cos_sched_rcv(g->sched_rcv, rfl, timeout,
 						&rcvd, &tid, &blocked, &cycles, &thd_timeout);
 			if (!tid) goto pending_events;
```

Comment:
likely prefer a `if (tid) {...}` here instead.  The goto feels a little messy

Reply:
yes, it makes sense. i'll remember to make this change too in my next PR.